File: fixtures/scale/java/medium/src/calc/repl/CommandHandler.java
package calc.repl;

import java.io.PrintStream;

import calc.history.HistoryFormatter;

public class CommandHandler {
    private final Session session;

    public CommandHandler(Session session) {
        this.session = session;
    }

    /** Returns false when the REPL should exit. */
    public boolean handle(String command, PrintStream out) {
        switch (command.trim()) {
            case ":quit":
            case ":q":
                return false;
            case ":vars":
                session.getEnvironment().names().forEach(out::println);
                return true;
            case ":history":
                out.println(HistoryFormatter.format(session.getHistory()));
                return true;
            case ":degrees":
                session.getSettings().useDegrees();
                return true;
            case ":radians":
                session.getSettings().useRadians();
                return true;
            default:
                out.println("unknown command " + command);
                return true;
        }
    }
}

File: fixtures/scale/dart/medium/lib/repl/command_handler.dart
import '../history/history_formatter.dart';
import 'session.dart';

class CommandHandler {
  final Session session;

  CommandHandler(this.session);

  /// Returns false when the REPL should exit.
  bool handle(String command) {
    switch (command.trim()) {
      case ':quit':
      case ':q':
        return false;
      case ':vars':
        session.environment.names().forEach(print);
        return true;
      case ':history':
        print(formatHistory(session.history));
        return true;
      case ':degrees':
        session.settings.useDegrees();
        return true;
      default:
        print('unknown command $command');
        return true;
    }
  }
}
